src/core/providers/ogr/qgsogrfeatureiterator.cpp

```diff
@@ -596,23 +597,39 @@ bool QgsOgrFeatureIterator::readFeature( const gdal::ogr_feature_unique_ptr &fet
   }
 
   // fetch attributes
+  const int fieldCount = mSource->mFields.count();
+  QgsAttributes attributes( fieldCount );
+  QVariant *attributeData = attributes.data();
   if ( mRequest.flags() & QgsFeatureRequest::SubsetOfAttributes )
   {
-    QgsAttributeList attrs = mRequest.subsetOfAttributes();
-    for ( QgsAttributeList::const_iterator it = attrs.constBegin(); it != attrs.constEnd(); ++it )
+    const int requestedAttributeTotal = mRequestAttributes.size();
+    int fetchedAttributes = 0;
+    if ( requestedAttributeTotal > 0 )
     {
-      getFeatureAttribute( fet.get(), feature, *it );
+      const int *requestAttribute = mRequestAttributes.constData();
```

Comment:
wouldn't something like the following (untested!) be simpler and slightly more efficient ?
```
for( int i = 0; i < requestedAttributeTotal; ++i )
{
    const int idx = requestAttribute[i];
    attributeData[idx] = getFeatureAttribute( fet.get(), idx );
}
```

Reply:
Indeed, much nicer! Thanks!